client/src/app/users/user-profile.component.ts

```diff
@@ -54,9 +55,11 @@ export class UserProfileComponent implements OnInit, OnDestroy {
         return user;
       },
       error: _err => {
-        this.snackBar.open('Problem loading the user – try again', 'OK', {
-          duration: 5000,
-        });
+        this.error = {
+          help: 'There was a problem loading the user – try again.',
+          httpResponse: _err.message,
+          message: _err.error ? _err.error.title : '',
```

Comment:
It took me quite a while to figure out how to get at the message we added in the Javalin code. It turns out it was in the `title` field in the `error` object. In the end I figured that out by (at the suggestions of esitsu@Twitch) printing `JSON.stringify(_err.error)`, which showed me to the structure of that object and its fields. Nothing I had searched gave me any indication where that message was, so using `JSON.stringify()` was a real lifesaver.

Reply:
Is it a bad thing that we are not looking at an error code? Or, did I just miss something and you do have an error code in there? I see that the error code 400 is displayed. Is the title where that comes from and it's not a special part of the error?

Reply:
Actually, the "400" that is displayed is in the `_err.message`, which is the HTTP response message the client receives.

The `_err.error` object also contains a `status` field that is `"400"`. So we could have a test that looks at `_err.error.status` and provides different messages depending on the status. I actually thought about doing that, but that sounds complicated to do _well_ and I'd rather this PR not get bogged down in that given how close we are to the start of the semester. I made an issue (#1451) for it, though, and we can bump up its priority if you think it's important.